src/main/java/com/kodilla/ecommercee/service/GroupService.java

```diff
@@ -0,0 +1,43 @@
+package com.kodilla.ecommercee.service;
+
+
+import com.kodilla.ecommercee.domain.Group;
+
+import com.kodilla.ecommercee.exception.GroupException;
+import com.kodilla.ecommercee.repository.GroupRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class GroupService {
+
+    private final GroupRepository groupRepository;
+
+    public List<Group> getAllGroups() {
+        return (List<Group>) groupRepository.findAll();
+    }
+
+    public Group getGroupById(Long id) throws GroupException {
+        return groupRepository.findById(id).orElseThrow(() -> new GroupException(
```

Comment:
throws GroupNotFoundException will be better. Now we don't know what is the problem with Group.

Reply:
okk, changed